tests/conftest.py

```diff
@@ -129,28 +131,88 @@ def _test_images_s3_bucket():
 MICROVM_S3_FETCHER = MicrovmImageS3Fetcher(_test_images_s3_bucket())
 
 
+# This should not be here. We should move ResultsFileDumper to the
+# framework directory since it is an auxiliary class used in the
+# integration tests. But we will do it in a separate commit.
 class ResultsFileDumper:  # pylint: disable=too-few-public-methods
     """Class responsible with outputting test results to files."""
 
-    def __init__(self, test_name: str, append=True):
+    def __init__(self, request, dump_results, generate_baselines):
         """Initialize the instance."""
-        if not append:
-            flags = "w"
-        else:
-            flags = "a"
+        test_name = request.node.originalname
+        baselines_file = request.param
 
         self._root_path = defs.TEST_RESULTS_DIR
+        self._results_file = None
+        self._baselines_file = None
 
         # Create the root directory, if it doesn't exist.
         self._root_path.mkdir(exist_ok=True)
 
-        self._file = open(self._root_path / test_name, flags)
-
-    def writeln(self, data: str):
-        """Write the `data` string to the output file, appending a newline."""
-        self._file.write(data)
-        self._file.write("\n")
-        self._file.flush()
+        if dump_results:
+            self._results_file = os.path.join(
+                self._root_path, "{}_results_{}.json".format(
+                    test_name, utils.get_host_kernel_version()))
+        if generate_baselines:
+            self._baselines_file = os.path.join(
+                self._root_path, "{}_config_{}.json".format(
+                    test_name, utils.get_host_kernel_version()))
+            open(self._baselines_file, "w+")
+            shutil.copy(baselines_file, self._baselines_file)
+
+    @staticmethod
+    def __dump_pretty_json(file, data, flags):
+        """Write the `data` dictionary to the output file in pretty format."""
+        with open(file, flags) as file_fd:
+            json.dump(data, file_fd, indent=4)
+            file_fd.write("\n")  # Add newline cause Py JSON does not
+
+    # pylint: disable=R1702
+    def __gen_baselines(self, result_dict):
```

Comment:
Hm, but I see no modifications to the baseline parsers. I believe they need some updates to keep them running with the latest changes (given the new test names and possibly the new dimension added by the host kernel version).
Do you also think it makes sense to add said flag to them that only updates the existing baselines instead of regenerating them altogether?